control-plane/subcommand/install-cni/binary.go

```diff
@@ -0,0 +1,56 @@
+package installcni
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// copyFile copies a file from a source directory to a destination directory.
+func copyFile(srcFile, destDir string) error {
+	// If the src file does not exist then either the incorrect command line argument was used or
+	// the docker container we built is broken somehow.
+	if _, err := os.Stat(srcFile); os.IsNotExist(err) {
+		return err
+	}
+
+	filename := filepath.Base(srcFile)
+	// If the destDir does not exist then the incorrect command line argument was used or
+	// the CNI settings for the kubelet are not correct.
+	info, err := os.Stat(destDir)
+	if os.IsNotExist(err) {
+		return err
+	}
+
+	if !info.IsDir() {
+		return fmt.Errorf("destination directory %s is not a directory", destDir)
+	}
+
+	// Check if the user bit is enabled in file permission.
+	if info.Mode().Perm()&(1<<(uint(7))) == 0 {
+		return fmt.Errorf("cannot write to destination directory %s", destDir)
```

Comment:
Might change this to say "user does not have permissions to..." so that this doesn't get confused with another failure mode.